src/infra/core/src/ingest/read_service.rs

```diff
@@ -61,6 +62,16 @@ impl ReadService {
             .provision_ingest_engine(listener.get_engine_provisioning_listener())
             .await?;
 
+        let prev_checkpoint_helper = if let Some(cp_hash) = prev_checkpoint {
+            Some(
+                ObjectProcessingHelper::from(&cp_hash, dataset.as_checkpoint_repo())
```

Comment:
Same problems with the helper here - more harm than good, so let's crash on non-local for now.